sys/mips/rootdev.c

```diff
@@ -173,9 +174,12 @@ static int rootdev_attach(device_t *bus) {
   return bus_generic_probe(bus);
 }
 
-static bus_methods_t rootdev_bus_if = {
+static icu_methods_t rootdev_icu_if = {
```

Comment:
could you please help me to understand why do we want to have intr methods in separate struct?

Reply:
Because we are going to have bus drivers which do not manage any ICUs. ISA is going to have a driver separate from the PCI one, yet they share the same interrupt controller unit. Either GT PCI driver will reamain responsible for managing ICU state and any interrupt-related routines from ISA devices will be propagated up to it, or the whole logic will be moved into ISA driver, which would leave PCI without the ability to manage interrupts.

TLDR; there's no one-to-one relation between Interrupt controllers and buses.

Reply:
ok, thank you for the explanation